internal/model/puller.go

```diff
@@ -367,13 +398,12 @@ func (p *Puller) pullerIteration(ignores *ignore.Matcher) int {
 	// Wait for the finisherChan to finish.
 	doneWg.Wait()
 
-	for i := range deletions {
-		deletion := deletions[len(deletions)-i-1]
-		if deletion.IsDirectory() {
-			p.deleteDir(deletion)
-		} else {
-			p.deleteFile(deletion)
-		}
+	for _, file := range fileDeletions {
+		p.deleteFile(file)
```

Comment:
We get the file from `WithNeed` anyway, so I don't see a lot of benefit doing additional DB lookups? 
Are you worried that adding the value to the map copies the struct?